### PR TITLE
Update to rust-bitcoin v0.15

### PR DESCRIPTION
**Pull request by stevenroose**
There is a strange failure in one test that I can't figure out:

https://gist.github.com/stevenroose/8a1ad0a1b4eaf29f35d41c8aa0d7bd96

### CODE REVIEW

**Comment by ariard** [comment]
Issue is into create_chan_between_nodes_with_value_init, L3522-L3524, we return a Transaction but register an OutPoint generated from tx serialization. 
Seems to me that's coming from the set of changes on serialization stuff in rust-bitcoin 0.15

**Comment by stevenroose** [comment]
Using `.txid()` solved it.

**Comment by apoelstra** [comment]
Test failures are likely caused by https://github.com/rust-bitcoin/rust-bitcoin/pull/153 which change to use segwit serialization for 0-input transactions. But this code was always incorrect for any transaction with witnesses.

File: src/ln/channel.rs [hunk 1]
```diff
@@ -2,11 +2,10 @@ use bitcoin::blockdata::block::BlockHeader;
 use bitcoin::blockdata::script::{Script,Builder};
 use bitcoin::blockdata::transaction::{TxIn, TxOut, Transaction, SigHashType};
 use bitcoin::blockdata::opcodes;
-use bitcoin::util::hash::{Sha256dHash, Hash160};
+use bitcoin::util::hash::{BitcoinHash, Sha256dHash, Hash160};
 use bitcoin::util::bip143;
 use bitcoin::network;
-use bitcoin::network::serialize::{BitcoinHash, RawDecoder, RawEncoder};
```
Comment (TheBlueMatt) [inline src/ln/channel.rs]: The bitcoin::network import above is now unused.

File: src/ln/channelmanager.rs [hunk 2]
```diff
@@ -3224,13 +3223,12 @@ mod tests {
...
 
-	use bitcoin::util::hash::Sha256dHash;
+	use bitcoin::util::hash::{BitcoinHash, Sha256dHash};
 	use bitcoin::blockdata::block::{Block, BlockHeader};
 	use bitcoin::blockdata::transaction::{Transaction, TxOut};
 	use bitcoin::blockdata::constants::genesis_block;
 	use bitcoin::network::constants::Network;
-	use bitcoin::network::serialize::serialize;
-	use bitcoin::network::serialize::BitcoinHash;
+	use bitcoin::consensus::encode::serialize;
```
Comment (TheBlueMatt) [inline src/ln/channelmanager.rs]: With the txid() change this is unused.

**Comment by stevenroose** [comment]
fixed and rebased

**Pull request merged**